daemon/cmd/agenthealth.go

```diff
@@ -73,18 +78,36 @@ func (d *Daemon) startAgentHealthHTTPService(addr string) {
 		}
 		w.WriteHeader(statusCode)
 	}))
-	srv := &http.Server{
-		Addr:    addr,
-		Handler: mux,
-	}
 
-	go func() {
-		err := srv.Serve(ln)
-		if errors.Is(err, http.ErrServerClosed) {
-			log.Info("healthz status API server shutdown")
+	available := len(hosts)
+	for _, host := range hosts {
+		lc := net.ListenConfig{Control: setsockoptReuseAddrAndPort}
+		addr := net.JoinHostPort(host, fmt.Sprintf("%d", option.Config.AgentHealthPort))
+		ln, err := lc.Listen(context.Background(), "tcp", addr)
+		if errors.Is(err, unix.EADDRNOTAVAIL) {
+			log.Infof("healthz status API server not available on %s", addr)
+			available--
+			continue
 		} else if err != nil {
-			log.WithError(err).Fatal("Unable to start healthz status API server")
+			log.WithError(err).Fatalf("Unable to listen on %s for healthz status API server", addr)
 		}
-	}()
-	log.Infof("Started healthz status API server on address %s", addr)
+
+		go func(addr string, ln net.Listener) {
+			srv := &http.Server{
+				Addr:    addr,
+				Handler: mux,
+			}
+			err := srv.Serve(ln)
+			if errors.Is(err, http.ErrServerClosed) {
+				log.Info("healthz status API server shutdown")
+			} else if err != nil {
+				log.WithError(err).Fatalf("Unable to start healthz status API server on address %s", addr)
+			}
+		}(addr, ln)
+		log.Infof("Started healthz status API server on address %s", addr)
```

Comment:
These logs should use structured logging, not the format variant on the logger. For example, `log.WithFields(...).Info(...)`, not `log.Infof(...)`.

Ideally all of the above should include the address in the logfields so that if there's multiple served then the logs are not duplicated, but have different parameters.

Reply:
Good point, thanks. Changed these to use `log.WithFields(...).Info(...)`.